pkg/apis/navigator/validation/generic.go

```diff
@@ -28,29 +30,61 @@ func ValidateImageSpec(img *navigator.ImageSpec, fldPath *field.Path) field.Erro
 		img.PullPolicy != corev1.PullIfNotPresent &&
 		img.PullPolicy != corev1.PullAlways &&
 		img.PullPolicy != "" {
-		el = append(el, field.NotSupported(fldPath.Child("pullPolicy"), img.PullPolicy, supportedPullPolicies))
+		el = append(
+			el,
+			field.NotSupported(fldPath.Child("pullPolicy"), img.PullPolicy, supportedPullPolicies),
+		)
 	}
 	return el
 }
 
-func ValidateNavigatorClusterConfig(cfg *navigator.NavigatorClusterConfig, fldPath *field.Path) field.ErrorList {
+var emptySemver = semver.Version{}
+
+func ValidateNavigatorClusterConfig(
+	cfg *navigator.NavigatorClusterConfig,
+	fldPath *field.Path,
+) field.ErrorList {
 	allErrs := ValidateImageSpec(&cfg.PilotImage, fldPath.Child("pilotImage"))
-	allErrs = append(allErrs, ValidateNavigatorSecurityContext(&cfg.SecurityContext, fldPath.Child("securityContext"))...)
+	if cfg.Image != nil {
+		allErrs = append(
+			allErrs,
+			ValidateImageSpec(cfg.Image, fldPath.Child("image"))...,
+		)
+	}
+	allErrs = append(
+		allErrs,
+		ValidateNavigatorSecurityContext(&cfg.SecurityContext, fldPath.Child("securityContext"))...,
+	)
 	sysctlsPath := fldPath.Child("sysctls")
 	for i, val := range cfg.Sysctls {
 		parts := strings.Split(val, "=")
 		if len(parts) != 2 {
-			allErrs = append(allErrs, field.Invalid(sysctlsPath.Index(i), val, "should be in format vm.max_map_count=65536"))
+			allErrs = append(
+				allErrs,
+				field.Invalid(sysctlsPath.Index(i), val, "should be in format vm.max_map_count=65536"),
+			)
 		}
 	}
+	if cfg.Version.Equal(emptySemver) {
+		allErrs = append(
+			allErrs,
+			field.Required(fldPath.Child("version"), "must be a semver version"),
+		)
+	}
 	return allErrs
 }
 
-func ValidateNavigatorSecurityContext(ctx *navigator.NavigatorSecurityContext, fldPath *field.Path) field.ErrorList {
+func ValidateNavigatorSecurityContext(
+	ctx *navigator.NavigatorSecurityContext,
+	fldPath *field.Path,
+) field.ErrorList {
 	el := field.ErrorList{}
 	if ctx.RunAsUser != nil {
 		if *ctx.RunAsUser < 0 {
-			el = append(el, field.Invalid(fldPath.Child("runAsUser"), *ctx.RunAsUser, "must be non-negative"))
+			el = append(
+				el,
+				field.Invalid(fldPath.Child("runAsUser"), *ctx.RunAsUser, "must be non-negative"),
```

Comment:
Please re-collapse. These append lines become harder to read when like this - append is a fairly trivial action, and we're turning a simple "add this error" into a bigger block of code than the rest of the function combined.